libs/libc/string/lib_strndup.c

```diff
@@ -34,6 +34,47 @@
  * Public Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name: nx_strndup
+ *
+ * Description:
+ *   The strndup() function is equivalent to the strdup() function,
+ *   duplicating the provided 's' in a new block of memory allocated as
+ *   if by using kmm_malloc(), with the exception being that strndup() copies
+ *   at most 'size' plus one bytes into the newly allocated memory,
+ *   terminating the new string with a NUL character. If the length of 's'
+ *   is larger than 'size', only 'size' bytes will be duplicated. If
+ *   'size' is larger than the length of 's', all bytes in s will be
+ *   copied into the new memory buffer, including the terminating NUL
+ *   character. The newly created string will always be properly
+ *   terminated.
+ *
+ ****************************************************************************/
+
+FAR char *nx_strndup(FAR const char *s, size_t size)
+{
+  FAR char *news = NULL;
```

Comment:
Minor: `NULL` init is redundant since is overwritten unconditionally few lines below

Reply:
I removed the `NULL` initialization.